test/test_read.ml

```diff
@@ -12,7 +12,57 @@ let from_file () =
   Alcotest.(check Testable.yojson) __LOC__ Fixtures.json_value (Yojson.Safe.from_file input_file);
   Sys.remove input_file
 
+let unquoted_from_string () =
+  Alcotest.(check Testable.yojson)
+    __LOC__
+    Fixtures.unquoted_value
+    (Yojson.Safe.from_string Fixtures.unquoted_json)
+
+let unquoted_from_lexbuf () =
+  let lexbuf = Lexing.from_string {|{foo: null, bar: "hello"}|} in
+  let lexer_state = Yojson.init_lexer () in
+
+  let ident_expected expectation reference start len =
+    let identifier = String.sub reference start len in
+    Alcotest.(check string)
+      (Format.asprintf "Reference '%s' start %d len %d matches '%s'" reference start len expectation)
+      expectation
+      identifier;
+    ()
+  in
+  let skip_over f =
+    f lexer_state lexbuf
+  in
+  let map_f mapper f =
+    mapper lexer_state f lexbuf
+  in
+  let map_ident = map_f Yojson.Safe.map_ident in
+  let map_string = map_f Yojson.Safe.map_string in
+
+  skip_over Yojson.Safe.read_lcurl;
+  map_ident (ident_expected "foo");
+  skip_over Yojson.Safe.read_colon;
+  skip_over Yojson.Safe.read_space;
+  skip_over Yojson.Safe.read_null;
+  skip_over Yojson.Safe.read_comma;
+  skip_over Yojson.Safe.read_space;
+  map_ident (ident_expected "bar");
+  skip_over Yojson.Safe.read_colon;
+  skip_over Yojson.Safe.read_space;
```

Comment:
Could become:
```suggestion
  skip_over Yojson.Safe.read_lcurl;
  map_ident (ident_expected "foo");
  skip_over Yojson.Safe.read_colon;
  skip_over Yojson.Safe.read_space;
```

Reply:
I kinda liked to exercise the `read_*` functions, but you're right, these don't actually need to be there and they should probably just be tested in a separate test that would exercise more of the possible values, so I removed it.